### ISSUE TITLE
Documents rewrite changes uploaded via the desktop client

### ISSUE THREAD

**Issue by olunet** [owncloud/documents#447]
ownCloud 7.0.4 Community Edition and Desktop Client 1.6.1.

When an odt-document is edited on a Desktop and then updated via the Desktop Client it does appear in the owncloud. I can download it and see all changes. However, when edited via Documents, a previous version is opened.

**Comment by aproposnix** [comment]
I've been having the same issue for months. I've upgraded to OC 8 this week but the issue is still there. 
I'm currently using the 1.7.1 version of the desktop client.

**Comment by olunet** [comment]
This is a critical issue, as it concerns serious data losses (and conflicts at personal level).

The problem exists for ownCloud 7.0.4 Community Edition and Desktop Client 1.7.1. Tested at several documents, at several machines, at two ownCloud installations.

**Comment by aproposnix** [comment]
Is more information needed for this issue?

**Comment by milkomeda** [comment]
same problem here using owncloud 8.0.4 and documents 0.9.0 on ubuntu 12.04 and oc client 1.8.1. makes the whole app unusable

**Comment by fifh** [comment]
Same issue with owncloud 8.1.0, documents 0.10, oc clients 1.8.4
Actually this issue was already created here : 
https://github.com/owncloud/documents/issues/349

**Comment by PVince81** [comment]
@VicDeo any update ?

**Comment by RealRancor** [comment]
@PVince81 A possible reason was posted here: https://github.com/owncloud/documents/issues/349#issuecomment-53842505

Also #349 looks really like a dup of this issue here.

**Comment by DanScharon** [comment]
seems to be fixed by #528 (but I didn't test it)

**Comment by RealRancor** [comment]
@DanScharon Nope, #528 was just closed as a duplicate of https://github.com/owncloud/documents/issues/447 which is still open and unfixed.

**Comment by quenenni** [comment]
Hello,

Just to say this problem still exists in oc v8.1.3,  documents v0.10.2 & oc client v2.0.1.
(server: Debian wheezy - Php 5.4.45-0+deb7u1 - Apache v2.2.22-13+deb7u6)

The things to do to test that:
- have a new file created in the files app or sync via the oc client,
- view, edit or download it shows the right content,
- share it (public share link) shows the right content.

Then, the things working well:
- modify it inside oc (via the document app) and closing it.
At that point, view, download or edit it (in your oc instance or via public share link) shows the new version.

But, the thing that creates the problem:
- modify it on your computer and sync it via the oc client
At that point, view or download it from the files app gives the new version
Edit it (in your oc instance or via public share link) shows the old version.

I tried things like removing everything related to cache in my apache server (expires module, cache in vhost file), I flushed memcached (echo 'flush_all' | nc localhost 11211), but the problem persisted.

So it seems that when you modify a document via the document app and the you close it, it executes routines that aren't called when you synch a modified file via the oc client.
That's all I could think of.

Hope this help.

PS: I didn't test with a webdav mount point.

**Comment by RealRancor** [comment]
Yes, this issue still exist as it is still open

**Comment by FreeMinded** [comment]
I can confirm this issue for oC 8.2.0 with documents 0.11.0
ODT edit on the desktop get synced fine. The viewer of the file app show correct version. In the documents app old version is displayed and any changes overwrite the file. Exact same behavior as  described by @quenenni.

**Comment by robpearman** [comment]
Still a problem on OC 9.0.1 and documents 0.12.0
